base/operators.jl

```diff
@@ -316,7 +316,7 @@ to_index(c::Colon) = c
 to_index(I::AbstractArray{Bool}) = find(I)
 to_index(A::AbstractArray{Int}) = A
 to_index{T<:Integer}(A::AbstractArray{T}) = [to_index(x) for x in A]
-to_index(i) = error("invalid index: $i")
+to_index(i) = i
```

Comment:
I suppose doing it this way makes for some less intelligible error messages.  We could be a little less lenient and keep the error, and add `to_index(i::AbstractArray) = i`.